fractions/javaee/jaxrs/src/main/java/org/wildfly/swarm/jaxrs/internal/JAXRSArchiveImpl.java

```diff
@@ -99,8 +112,62 @@ private static boolean hasApplicationPathAnnotation(Archive<?> archive) {
         return false;
     }
 
+    private static boolean hasApplicationServletMapping(Archive<?> archive) {
+        Node webXmlNode = archive.get(PATH_WEB_XML);
+        if (webXmlNode != null) {
+            return hasApplicationServletMapping(webXmlNode.getAsset());
+        }
+        return false;
+    }
+
+    private static boolean hasApplicationServletMapping(Asset webXmlAsset) {
```

Comment:
The WebXmlAsset has an accessor for testing if a servlet exists.  If we need to add another accessor to test if a servlet-mapping exists, that cool.

I still prefer using the shrinkwrap descriptors, though, instead of XML/XPath to poke around, if you don't object.